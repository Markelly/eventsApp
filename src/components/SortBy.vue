<template>
  <div class="sort-by">
    <span class="sort-by__title">Sort by</span>
    <select
      id="orderBy"
      @change="sortEvents(sort)"
      v-model="sort"
      data-sort-select
    >
      <option
        v-for="(option, index) in sortOptions"
        :key="index"
        :value="option.value"
        v-text="option.text"
        data-sort-options
      />
    </select>
  </div>
</template>

<script>
import { SORT_OPTIONS } from "../utils/constants";
export default {
  name: "SortBy",
  data() {
    return {
      sort: "",
      sortOptions: SORT_OPTIONS
    };
  },
  methods: {
    sortEvents(sort) {
      this.$emit("sortEvents", sort);
    }
  }
};
</script>

<style scoped lang="scss">
@import "../styles/events-app";
.sort-by {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  font-size: 12px;
  padding: 16px 8px;
  color: $color6;

  &__title {
    padding-right: 8px;
  }

  #orderBy {
    height: 30px;
    width: 120px;
    background: $color2;
    color: $color6;

    &:hover {
      border: solid 1px $color6;
      background: $color1;
      cursor: pointer;
    }
  }
}
</style>
