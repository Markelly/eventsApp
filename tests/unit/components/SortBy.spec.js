import SortBy from "@/components/SortBy";
import { createLocalVue, shallowMount } from "@vue/test-utils";
import BootstrapVue from "bootstrap-vue";

const localVue = createLocalVue();
localVue.use(BootstrapVue)

describe("<SortBy/>", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(SortBy, {
      localVue
    });
  });

  test("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should emit sortEvents event", () => {
    wrapper.vm.sortEvents("sort");
    expect(wrapper.emitted("sortEvents", "sort")).toBeTruthy();
  });
});
