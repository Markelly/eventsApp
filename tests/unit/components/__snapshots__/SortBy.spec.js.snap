// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortBy/> should render the component 1`] = `
<div class="sort-by ml-auto">
  <b-form-select-stub options="[object Object],[object Object],[object Object],[object Object],[object Object]" valuefield="value" textfield="text" htmlfield="html" disabledfield="disabled" labelfield="label" optionsfield="options" selectsize="0"></b-form-select-stub>
</div>
`;
